builder/format.go

```diff
@@ -0,0 +1,132 @@
+
+package builder
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/pkg/errors"
+)
+
+// UpdateFormatVersion updates the builder.conf file with a new format version
+func (b *Builder) UpdateFormatVersion(version string) error {
+	b.Config.Swupd.Format = version
+
+	newver := "${1}" + b.Config.Swupd.Format
+	var re = regexp.MustCompile(`(FORMAT=)[0-9]*`)
+
+	builderData, err := ioutil.ReadFile(b.BuildConf)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read builder.conf")
+	}
+
+	builderEdit := re.ReplaceAllString(string(builderData), newver)
+
+	var filename string
+	if UseNewConfig {
+		filename, err = GetConfigPath("")
+		if err != nil {
+			return err
+		}
+
+		return b.Config.SaveConfig(filename)
+	}
+
+	builderOut := []byte(builderEdit)
+	if err = ioutil.WriteFile(b.BuildConf, builderOut, 0644); err != nil {
+		return errors.Wrap(err, "Failed to write new builder.conf")
+	}
+
+	return nil
+}
+
+// CopyFullGroupsINI copies the initial ini file which has ALL bundle definitions
+func (b *Builder) CopyFullGroupsINI() error {
+	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "full_groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"))
+}
+
+// CopyTrimmedGroupsINI copies the new ini made with deleted bundles removed
+func (b *Builder) CopyTrimmedGroupsINI() error {
+	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "trimmed_groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"))
+}
+
+// RevertFullGroupsINI copies back the full ini to the manifest creator accounts for deleted bundles
+func (b *Builder) RevertFullGroupsINI() error {
+	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "full_groups.ini"))
+}
+
+// RevertTrimmedGroupsINI copies back the trimmed INI so manifests are not created anymore for deleted bundles in new format
+func (b *Builder) RevertTrimmedGroupsINI() error {
+	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "trimmed_groups.ini"))
+}
+
+func (b *Builder) getLastBuildVersion() (string, error) {
+	var lastVer []byte
+	var err error
+
+	filename := filepath.Join(b.Config.Builder.ServerStateDir, "image/LAST_VER")
+	// Likely the first build
+	if lastVer, err = ioutil.ReadFile(filename); os.IsNotExist(err) {
+		return "", nil
+	} else if err != nil {
+		return "", errors.Wrap(err, "Cannot find last built version")
+	}
+	data := string(lastVer)
+	ver := strings.Split(data, "\n")
+
+	return ver[0], nil
+}
+
+// CheckBumpNeeded returns nil if it successfully deduces there is no format
+// bump boundary being crossed.
+func (b *Builder) CheckBumpNeeded() (bool, error) {
+	version, err := b.getLastBuildVersion()
+	if err != nil {
+		return false, nil
```

Comment:
Wait, sorry this is going back and forth, but now that we're only returning an error from `getlastBuildVersion` if there _is_ an error, this needs to go back to `return false, err`.